client/src/components/itineraries/Comments.js

```diff
@@ -0,0 +1,201 @@
+  return comments.filter(
+    //When I post a new comment, it doesn't populate the itinerary field,
+    //so comment.itinerary will be just the id of that itinerary,
+    //thats why comment.itinerary._id get the id of the already populated,
+    //and comment.itinerary gets it for the ones just posted.
+    comment => (comment.itinerary._id || comment.itinerary) === itinerary._id
+  );
+};
+
+export class Comments extends Component {
+  constructor(props) {
+    super(props);
+    this.textInput = React.createRef();
+    this.onClickHandler = this.onClickHandler.bind(this);
+    this.updateInputValue = this.updateInputValue.bind(this);
+    this.state = {
+      inputValue: "",
+      comments: []
+    };
+  }
+
+  //componentWillReceiveProps is deprecated
+  UNSAFE_componentWillReceiveProps(nextProps) {
+    if (nextProps.comments.payload !== this.props.comments.payload) {
+      //even though I send the comments already sorted from back, it doesn't work properly
+      let comments = Array.from(nextProps.comments.payload);
+      function compare(a, b) {
+        if (a.date < b.date) return 1;
+        if (a.date > b.date) return -1;
+        return 0;
+      }
+      comments.sort(compare);
+      //call the function that gets the comments for that specific Itinerary
+      comments = findComments(nextProps.itinerary, comments);
+      this.setState({
+        comments: comments
+      });
+      //this querySelector inside the if prevents the toast to appear if there is another
+      //toast active. It gets twice inside this else if so that's why I don't want
+      //a second toast for the same comment
+    } else if (
+      nextProps.comments.toast === true &&
+      !document.querySelector(".toast")
+    ) {
+      var toastHTML = `<span>Message sent</span>`;
+      M.toast({ html: toastHTML, classes: "rounded" });
+    }
+  }
+
+  onClickHandler() {
+    if (this.state.inputValue !== "") {
+      let itinerary = this.props.itinerary;
+      let commentBody = {
+        user: {
+          name: this.props.user.name,
+          photo: this.props.user.image
+        },
+        itineraryId: itinerary._id,
+        message: this.state.inputValue,
+        date: Date.now(),
+        city: itinerary.city._id
+      };
+      //calls the action that posts a comment
+      this.props.postComment(commentBody);
+      this.setState({
+        inputValue: ""
+      });
+    }
+  }
+
+  handleKeyUp = event => {
+    if (event.key === "Enter") {
+      this.onClickHandler();
+    }
+  };
+
+  updateInputValue() {
+    this.setState({
+      inputValue: this.textInput.current.value
+    });
+  }
+
+  formatDate(date) {
+    var hours = date.getHours();
+    var minutes = date.getMinutes();
+    var ampm = hours >= 12 ? "pm" : "am";
+    hours = hours % 12;
+    hours = hours ? hours : 12; // the hour '0' should be '12'
+    minutes = minutes < 10 ? "0" + minutes : minutes;
+    var strTime = hours + ":" + minutes + " " + ampm;
+    return (
```

Comment:
You might find it easier to use [template strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) here.